### ISSUE TITLE
Compile issue on gfortran 10.x

### ISSUE THREAD

**Issue by jagot** [jagot/bsr#4]
While compiling with gfortran 10.2.0, I got the following error:
```make
[ 83%] Building Fortran object dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/gen_conf.f90.o
/Users/jagot/prog/bsr/src/dbsr/DBSR_CONF3/gen_conf.f90:132:51:

  132 |         if(Jintra(no).eq.Jtotal) Call Check_cfg(ii)
      |                                                   1
Error: More actual than formal arguments in procedure call at (1)
make[2]: *** [dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/gen_conf.f90.o] Error 1
make[1]: *** [dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/all] Error 2
make: *** [all] Error 2
```
If I instead force the gfortran version by `FC=gfortran-9 cmake && make`, the error disappears. I think this is related to a general tightening of previously lax error handling in gfortran, similar to what we see in [Grasp](https://github.com/compas/grasp/issues/46).

@katrham20 Have you seen this? Forcing the gfortran version to 9 of course helps in the short run, but I guess we really want to make the code standards-compliant in the long run.

**Comment by katrham20** [comment]
Hi Stefanos,

Oh I had not seen this - I had mostly been working on tidying up things with the Intel compiler but have not looked at this in a while. Will keep this in mind!

On 19 Oct 2020, at 02:29, Stefanos Carlström <notifications@github.com<mailto:notifications@github.com>> wrote:

This message is from an external sender. Please take care when responding, clicking links or opening attachments.


While compiling with gfortran 10.2.0, I got the following error:

[ 83%] Building Fortran object dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/gen_conf.f90.o
/Users/jagot/prog/bsr/src/dbsr/DBSR_CONF3/gen_conf.f90:132:51:

  132 |         if(Jintra(no).eq.Jtotal) Call Check_cfg(ii)
      |                                                   1
Error: More actual than formal arguments in procedure call at (1)
make[2]: *** [dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/gen_conf.f90.o] Error 1
make[1]: *** [dbsr/DBSR_CONF3/CMakeFiles/dbsr_conf3.dir/all] Error 2
make: *** [all] Error 2

If I instead force the gfortran version by FC=gfortran-9 cmake && make, the error disappears. I think this is related to a general tightening of previously lax error handling in gfortran, similar to what we see in Grasp<https://github.com/compas/grasp/issues/46>.

@katrham20<https://github.com/katrham20> Have you seen this? Forcing the gfortran version to 9 of course helps in the short run, but I guess we really want to make the code standards-compliant in the long run.

**Comment by jagot** [comment]
Fixed in 8df7b606f2624ba20f5dd327a874834e7e32cbb8 (cf https://github.com/zatsaroi/DBSR3/commit/b9d417ad2d3654092388488e47fdf8ee205071ae).
